test/libsolidity/SyntaxTest.cpp

```diff
@@ -132,8 +132,13 @@ void SyntaxTest::filterObtainedErrors()
 			// ignore the version & license pragma inserted by the testing tool when calculating locations.
 			if (location->start != -1)
 			{
-				solAssert(location->start >= preambleSize, "");
-				locationStart = location->start - preambleSize;
+				if (location->start >= preambleSize)
```

Comment:
It would be a good idea to add a comment explaining the weirdness happening here:
```suggestion
				// Preamble is not in the original input so error locations need to be shifted back.
				// Normally this is safe because errors should not happen inside preamble. Errors
				// covering the whole file are a special case because they start at 0, not after the preamble.
				if (location->start >= preambleSize)
```